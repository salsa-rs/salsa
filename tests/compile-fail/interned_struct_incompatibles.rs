#[salsa::interned(returns(ref))]
struct InternedWithRetRef {
    field: u32,
}

#[salsa::interned(specify)]
struct InternedWithSpecify {
    field: u32,
}

#[salsa::interned(no_eq)]
struct InternedWithNoEq {
    field: u32,
}

#[salsa::interned(db = Db)]
struct InternedWithDb {
    field: u32,
}

#[salsa::interned(recover_fn = recover)]
struct InternedWithRecover {
    field: u32,
}

#[salsa::interned(lru = 12)]
struct InternedWithLru {
    field: u32,
}

#[salsa::interned]
struct InternedWithTrackedField {
    #[tracked]
    field: u32,
}

#[salsa::interned(revisions = 0)]
struct InternedWithZeroRevisions {
    field: u32,
}

fn main() {}
