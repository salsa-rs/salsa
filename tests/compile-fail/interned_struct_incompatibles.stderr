error: `returns` option not allowed here
 --> tests/compile-fail/interned_struct_incompatibles.rs:1:19
  |
1 | #[salsa::interned(returns(ref))]
  |                   ^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/interned_struct_incompatibles.rs:6:19
  |
6 | #[salsa::interned(specify)]
  |                   ^^^^^^^

error: `no_eq` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:11:19
   |
11 | #[salsa::interned(no_eq)]
   |                   ^^^^^

error: `db` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:16:19
   |
16 | #[salsa::interned(db = Db)]
   |                   ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/interned_struct_incompatibles.rs:21:19
   |
21 | #[salsa::interned(recover_fn = recover)]
   |                   ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:26:19
   |
26 | #[salsa::interned(lru = 12)]
   |                   ^^^

error: `#[tracked]` cannot be used with `#[salsa::interned]`
  --> tests/compile-fail/interned_struct_incompatibles.rs:33:5
   |
33 | /     #[tracked]
34 | |     field: u32,
   | |______________^

error: cannot find attribute `tracked` in this scope
  --> tests/compile-fail/interned_struct_incompatibles.rs:33:7
   |
33 |     #[tracked]
   |       ^^^^^^^
   |
help: consider importing one of these attribute macros
   |
1  + use salsa::tracked;
   |
1  + use salsa_macros::tracked;
   |
